### PR TITLE
Refactor HTableNameConverter

### PR DESCRIPTION
**Pull request by chtyim**
- Get rid of unnecessary static
  - Static is really evil
- Cleanup code, remove unneeded methods
- Simplify the HTableNameConverter and sub-classes

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HTableNameConverter.java [hunk 1]
```diff
@@ -67,78 +67,72 @@ private static String getBackwardCompatibleTableName(String tablePrefix, TableId
...
       String prefix = parts[0];
-      hTableName = parts[1];
-      return PrefixedTableId.from(prefix, namespace, hTableName);
+      qualifier = parts[1];
+      return new PrefixedTableId(prefix, namespace, qualifier);
     }
 
-
-    String[] parts = hBaseNamespace.split("_");
+    // If HBase namespace is used, namespace is something like 'cdap_userNS'
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HTableNameConverter.java]: This should probably say "If non-default namespace is used..."

**Comment by anwar6953** [comment]
remainder lgtm

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HTableNameConverter.java (hunk 1 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HTableNameConverter.java]: ok

**Comment by chtyim** [comment]
Build running in http://builds.cask.co/browse/CDAP-DUT1050-1

**Comment by chtyim** [comment]
Build passed. http://builds.cask.co/browse/CDAP-DUT1050-5

**Pull request merged**